src/codegen/compile.ml

```diff
@@ -5544,15 +5544,14 @@ module MakeSerialization (Strm : Stream) = struct
       (* Sanity check: Did we fill exactly the buffer *)
       get_refs_start ^^ get_refs_size ^^ compile_mul_const Heap.word_size ^^ G.i (Binary (Wasm.Values.I32 I32Op.Add)) ^^
       G.i (Compare (Wasm.Values.I32 I32Op.Eq)) ^^
-      E.else_trap_with env "reference buffer not filled " ^^
+      E.else_trap_with env "reference buffer not filled" ^^
 
       get_data_start ^^ get_data_size ^^ G.i (Binary (Wasm.Values.I32 I32Op.Add)) ^^
       G.i (Compare (Wasm.Values.I32 I32Op.Eq)) ^^
-      E.else_trap_with env "data buffer not filled " ^^
+      E.else_trap_with env "data buffer not filled" ^^
 
       get_refs_size ^^
-      compile_unboxed_const 0l ^^
-      G.i (Compare (Wasm.Values.I32 I32Op.Eq)) ^^
+      G.i (Test (Wasm.Values.I32 I32Op.Eqz)) ^^
```

Comment:
Using
```
compile_eq_const 0l
```
and optimizing there or in the peep-hole-optimizier might be a bit more declarative and less line-noisy.

Reply:
Yes, #3149 already does the emissions part. I will prepare a peephole patch when I have time.

Reply:
I left https://github.com/dfinity/motoko/issues/3183 for the peephole.